core/capabilities/ccip/oraclecreator/plugin.go

```diff
@@ -438,17 +373,16 @@ func (i *pluginOracleCreator) createReadersAndWriters(
 			return nil, nil, fmt.Errorf("failed to get chain selector from chain ID %s: %w", chainID, err1)
 		}
 
-		if _, exists := plugins[relayChainFamily]; !exists {
-			i.lggr.Debugw("createReadersAndWriters: skipping unsupported relayer", "chainID", chainID, "family", relayChainFamily)
-			continue
-		}
-
-		chainReaderConfig, err1 := getChainReaderConfig(i.lggr, chainID, destChainID, homeChainID, ofc, chainSelector, relayChainFamily)
-		if err1 != nil {
-			return nil, nil, fmt.Errorf("failed to get chain reader config: %w", err1)
-		}
-
-		cr, err1 := relayer.NewContractReader(ctx, chainReaderConfig)
+		cr, err1 := crcw.GetChainReader(ctx, ccipcommon.GetChainReaderParams{
+			Lggr:             i.lggr,
+			Relayer:          relayer,
+			ChainID:          chainID,
+			DestChainID:      destChainID,
```

Comment:
Isn't the problem here that destChainID is not the chain for chainSelector param?
Coz later in this function the code this in L423:
contractReaders[chainSelector] = cr